ui/lesson-renderer/src/main/java/org/cru/godtools/tool/lesson/ui/LessonActivity.kt

```diff
@@ -80,7 +82,7 @@ class LessonActivity :
     // region Pages
     @Inject
     lateinit var lessonPageAdapterFactory: LessonPageAdapter.Factory
-    private val lessonPageAdapter by lazy { lessonPageAdapterFactory.create(this, this) }
+    private val lessonPageAdapter by lazy { lessonPageAdapterFactory.create(this, this, toolState.toolState) }
```

Comment:
```by lazy``` simply calls the method when ```get()``` first runs and then returns the result of that first call if ```get()``` gets called again? Is that to just prevent unnecessary calls to that method?  And if i'm understanding the syntax correctly(I could be way off here haha):
```kotlin
private val lessPageAdapter = lessPageAdapterFactory.create(this, this, toolState.toolState)
```
is the same as 
```kotlin
private val lessonPageAdapter
        get() = lessonPageAdapterFactory.create(this, this, toolState.toolState)
```
which is the same as the ```by lazy``` syntax except for the additional things mentioned above?

Reply:
you got the syntax right. We want to reuse the same `lessonPageAdapter` for the lifetime of the Activity, which is the reason we don't use `get() =`. But we also can't initialize the adapter on object creation because `lessonPageAdapterFactory` won't be initialized, plus the parameters sent to `create` won't be in the correct state until the activity is actually started.

Reply:
Sweet, makes sense! I feel like the syntax is beginning to click more and more.